src/Dom/Document.php

```diff
@@ -666,6 +668,45 @@ private function maybe_restore_noscript_elements( $html ) {
 		);
 	}
 
+	/**
+	 * Replaces the tag name of script[template="amp-mustache"], as a workaround to a parsing issue.
+	 *
+	 * This script can have closing tags of its children table and td stripped.
+	 * So this changes its name from script to tmp-script to avoid this.
+	 *
+	 * @link https://github.com/ampproject/amp-wp/issues/4254
+	 * @see restore_mustache_templates() Reciprocal function.
+	 *
+	 * @param string $html To replace the tag name of the mustache templates in.
+	 * @return string The HTML, with the tag name of the mustache templates replaced.
+	 */
+	private function replace_mustache_templates( $html ) {
+		return preg_replace(
+			'#<script(\s[^>]*template=["\']?amp-mustache["\']?[^>]*)>(.*?)</script>#s',
```

Comment:
I'd recommend the following changes to the regex to make it less brittle:
- Make it case-insensitive by adding the `i` flag
- Support whitespace at the end of the closing tag (`</script >`), which is valid HTML
- Only match a pair of matching quotes, instead of a mismatch, which would be parsed differently
- Make the initial non-closing-bracket group non-greedy to avoid some backtracking (`<script(\s[^>]*?`)

Reply:
Sure, applied in https://github.com/ampproject/amp-wp/pull/4333/commits/52d49e93327b7f4ae041e6dd490e0b1ebdbf3ece